Framework/Indexing/inc/MantidIndexing/DetectorID.h

```diff
@@ -34,16 +33,13 @@ namespace Indexing {
   File change history is stored at: <https://github.com/mantidproject/mantid>
   Code Documentation is available at: <http://doxygen.mantidproject.org>
 */
-template <class T1, class T2,
-          class = typename std::enable_if<std::is_integral<T1>::value &&
-                                          std::is_integral<T2>::value>::type>
-std::vector<T2> makeRange(T1 first, T2 last) {
-  std::vector<T2> vec(last - static_cast<T2>(first) + 1);
-  std::iota(vec.begin(), vec.end(), static_cast<T2>(first));
-  return vec;
-}
+struct MANTID_INDEXING_DLL DetectorID
```

Comment:
@OwenArnold Yes, up to a point. Conversion from `DetectorID -> int` is explicit, but I am currently allowing implicit conversion from `int -> DetectorID`. I initially forbid both, but it lead to a lot of bloat.